builtin/logical/database/path_creds_create.go

```diff
@@ -158,14 +206,22 @@ func (b *databaseBackend) pathStaticCredsRead() framework.OperationFunc {
 			return nil, fmt.Errorf("%q is not an allowed role", name)
 		}
 
+		respData := map[string]interface{}{
+			"username":            role.StaticAccount.Username,
+			"ttl":                 role.StaticAccount.CredentialTTL().Seconds(),
+			"rotation_period":     role.StaticAccount.RotationPeriod.Seconds(),
+			"last_vault_rotation": role.StaticAccount.LastVaultRotation,
+		}
+
+		switch role.CredentialType {
+		case v5.CredentialTypePassword:
```

Comment:
We should also account for the `""` case and default this to the password behavior for existing, older roles. This should be applicable to all cases with similar switch on the role's credential type.

Reply:
Similar comment above. Don't need to check for "" as it's an int.